ratis-client/src/main/java/org/apache/ratis/client/api/AsyncApi.java

```diff
@@ -40,6 +40,8 @@
    */
   CompletableFuture<RaftClientReply> send(Message message);
 
+  CompletableFuture<RaftClientReply> sendForward(Message message);
```

Comment:
Similar to DataStreamRpcApi, we should add an AsyncRpcApi so that the new method could be hidden from users.

The parameter should be a RaftClientRequest.
```
  public CompletableFuture<RaftClientReply> sendForward(RaftClientRequest request);
```